### PR TITLE
Implement GetBlockHeaderMerkleParts and test

### LINKED ISSUE #1323: Implement GetBlockHeaderMerkleParts and test
Implement function Commit (use Header) -> BlockHeaderMerkleParts + test Commit -> BlockHeaderMerkleParts + blockheight + apphash -> Blockhash and compare with blockhash in commit

### PR DESCRIPTION
**Pull request by taobun**
fixed #1323

### CODE REVIEW

File: chain/go.sum [hunk 1]
```diff
@@ -2,14 +2,32 @@ cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMT
 cloud.google.com/go v0.34.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
 github.com/99designs/keyring v1.1.4 h1:x0g0zQ9bQKgNsLo0XSXAy1H8Q1RG/td+5OXJt+Ci8b8=
 github.com/99designs/keyring v1.1.4/go.mod h1:657DQuMrBZRtuL/voxVyiyb6zpMehlm5vLB9Qwrv904=
+github.com/Azure/azure-pipeline-go v0.2.1/go.mod h1:UGSo8XybXnIGZ3epmeBw7Jdz+HiUVpqIlpz/HKHylF4=
```
Comment (sorawit) [inline chain/go.sum]: Did you make sure to run go mod tidy?

File: chain/client/proof.go [hunk 2]
```diff
@@ -3,537 +3,526 @@ package rpc
 // TODO: Revive this! It's commented because we upgrade to Cosmos 0.38.*.
```
Comment (sorawit) [inline chain/client/proof.go]: Let's remove the TODO if this is done?

File: chain/client/proof.go [hunk 3]
```diff
@@ -3,537 +3,526 @@ package rpc
...
+	bp.SignedDataPrefix = kb
+
+	if err != nil {
+		return BlockRelayProof{}, err
+	}
+
+	return bp, nil
+}
+
+func MakeOtherStoresMerkleHash(mspo rootmulti.MultiStoreProofOp) (tmbytes.HexBytes, tmbytes.HexBytes, tmbytes.HexBytes) {
```
Comment (sorawit) [inline chain/client/proof.go]: Can we break this down into smaller, more understandable chunks that we can test separately?

File: chain/client/proof.go [hunk 4]
```diff
@@ -3,537 +3,526 @@ package rpc
...
+	var oracleHash []byte
+	m := map[string][]byte{}
+	for _, si := range mspo.Proof.StoreInfos {
+		m[si.Name] = tmhash.Sum(tmhash.Sum(si.Core.CommitID.Hash))
+		if si.Name == "oracle" {
```
Comment (sorawit) [inline chain/client/proof.go]: Can we take this string from some defined variable? `oracle.ModuleName`?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/go.sum (hunk 1 above)
Comment (taobun) [reply chain/go.sum]: yep

File: chain/client/proof.go [hunk 5]
```diff
@@ -1,322 +1,154 @@
...
+	return IAVLMerklePathEthereum{
+		merklePath.IsDataOnRight,
+		merklePath.SubtreeHeight,
+		big.NewInt(int64(merklePath.SubtreeSize)),
+		big.NewInt(int64(merklePath.SubtreeVersion)),
+		common.BytesToHash(merklePath.SiblingHash),
+	}
+}
+
+type BlockHeaderMerkleParts struct {
```
Comment (sorawit) [inline chain/client/proof.go]: Let's move this into its own file

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by tansawit — APPROVED** [review]

**Pull request merged**